qiita_db/analysis.py

```diff
@@ -429,15 +429,17 @@ def mapping_file(self):
 
         Returns
         -------
-        str or None
-            full filepath to the mapping file or None if not generated
+        int or None
+            The filepath id of the analysis mapping file or None
+            if not generated
         """
-        fp = [fp for _, fp, fp_type in qdb.util.retrieve_filepaths(
-            "analysis_filepath", "analysis_id", self._id)
-            if fp_type == 'plain_text']
+        fp = [fp_id
+              for fp_id, _, fp_type in qdb.util.retrieve_filepaths(
+                "analysis_filepath", "analysis_id", self._id)
+              if fp_type == 'plain_text']
```

Comment:
It's accurate that a mapping file is "plain_text" but should the system be able to differentiate those types with, say, biom summary? Out of context for this PR but seemed unexpected

Reply:
At this point only the mapping file is stored in the table "analysis_filepath". The BIOM-related files (and others) are specifically stored with the artifact, not directly to the analysis.

Reply:
"At this point" suggests this could lead to bugs in the future if that changes?

Reply:
If the data model changes in the future then other sections beside this one will need to be updated - so I would say that it would depend on the nature of the data model changes.

Reply:
Discussing offline, @wasade and I agreed that perhaps the solution of this specific kind of conflicts will be to implement something like https://github.com/biocore/qiita/issues/2361